internal/provider/models/processors/data_profiler.go

```diff
@@ -0,0 +1,102 @@
+package processors
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	. "github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	. "github.com/mezmo/terraform-provider-mezmo/v4/internal/client"
+	. "github.com/mezmo/terraform-provider-mezmo/v4/internal/provider/models/modelutils"
+)
+
+const DATA_PROFILER_PROCESSOR_TYPE_NAME = "data_profiler"
+const DATA_PROFILER_PROCESSOR_NODE_NAME = "data-profiler"
+
+type DataProfilerProcessorModel struct {
+	Id           StringValue `tfsdk:"id"`
+	PipelineId   StringValue `tfsdk:"pipeline_id"`
+	Title        StringValue `tfsdk:"title"`
+	Description  StringValue `tfsdk:"description"`
+	Inputs       ListValue   `tfsdk:"inputs"`
+	GenerationId Int64Value  `tfsdk:"generation_id"`
+	AppFields    ListValue   `tfsdk:"app_fields" user_config:"true"`
+	HostFields   ListValue   `tfsdk:"host_fields" user_config:"true"`
+	LevelFields  ListValue   `tfsdk:"level_fields" user_config:"true"`
+	LineFields   ListValue   `tfsdk:"line_fields" user_config:"true"`
+	LabelFields  ListValue   `tfsdk:"label_fields" user_config:"true"`
+}
+
+var DataProfilerProcessorResourceSchema = schema.Schema{
+	Description: "Profile the data sent through your pipeline, generating annotations and saving profile data.",
+	Attributes: ExtendBaseAttributes(map[string]schema.Attribute{
+		"app_fields":   ListAttributeSchemaGenerator("app"),
+		"host_fields":  ListAttributeSchemaGenerator("host"),
+		"level_fields": ListAttributeSchemaGenerator("level"),
+		"line_fields":  ListAttributeSchemaGenerator("line"),
+		"label_fields": ListAttributeSchemaGenerator("label"),
+	}),
+}
+
+func ListAttributeSchemaGenerator(attributeName string) schema.ListAttribute {
+	return schema.ListAttribute{
```

Comment:
Should we make this optional and computed?
It looks like the API sets defaults if you don't specify anything

Reply:
Yeah I was gonna say the UI has a ton of defaults here

Reply:
I believe these are actually required in the API, just that it also provides UI hints to the frontend which are defaults there

Reply:
actually I was partly wrong, `label_fields` are not required. updated